qiskit/algorithms/optimizers/scipy_optimizer.py

```diff
@@ -0,0 +1,151 @@
+    See https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html
+    """
+
+    def __init__(
+        self,
+        method: Union[str, Callable],
+        tol: Optional[float] = None,
+        options: Optional[Dict[str, Any]] = None,
+        max_evals_grouped: int = 1,
+        **kwargs,
+    ):
+        """
+        Args:
+            method: Type of solver.
+            tol: Tolerance for termination.
+            options: A dictionary of solver options.
+            kwargs: additional kwargs for scipy.optimize.minimize.
+            max_evals_grouped: Max number of default gradient evaluations performed simultaneously.
+        """
+        # pylint: disable=super-init-not-called
+        self._method = method.lower() if isinstance(method, str) else method
+        self._set_support_level()
+        self._tol = tol
+        self._options = options if options is not None else {}
+        validate_min("max_evals_grouped", max_evals_grouped, 1)
+        self._max_evals_grouped = max_evals_grouped
+        self._kwargs = kwargs
+
+    def get_support_level(self):
+        """Return support level dictionary"""
+        return {
+            "gradient": self._gradient_support_level,
+            "bounds": self._bounds_support_level,
+            "initial_point": self._initial_point_support_level,
+        }
+
+    def _set_support_level(self):
+        if self._method in {"l-bfgs-b", "tnc", "slsqp", "powell", "trust-constr"}:
+            self._bounds_support_level = OptimizerSupportLevel.supported
+        else:
+            self._bounds_support_level = OptimizerSupportLevel.ignored
+
+        if self._method in {
+            "cg",
+            "bfgs",
+            "newton-cg",
+            "l-bfgs-b",
+            "tnc",
+            "slsqp",
+            "dogleg",
+            "trust-ncg",
+            "trust-krylov",
+            "trust-exact",
+            "trust-constr",
+        }:
+            self._gradient_support_level = OptimizerSupportLevel.supported
+        else:
+            self._gradient_support_level = OptimizerSupportLevel.ignored
+
+        self._initial_point_support_level = OptimizerSupportLevel.required
+
+    def optimize(
+        self,
+        num_vars,
+        objective_function,
+        gradient_function=None,
+        variable_bounds: Optional[Union[Sequence, Bounds]] = None,
+        initial_point=None,
+    ):
+        # Remove ignored parameters to supress the warning of scipy.optimize.minimize
+        if self.is_bounds_ignored:
+            variable_bounds = None
+        if self.is_gradient_ignored:
+            gradient_function = None
+
+        if self.is_gradient_supported and gradient_function is None and self._max_evals_grouped > 1:
+            if "eps" in self._options:
+                epsilon = self._options["eps"]
+            else:
+                epsilon = (
+                    1e-8 if self._method in {"l_bfgs_b", "tnc"} else np.sqrt(np.finfo(float).eps)
```

Comment:
Were these numbers the default before or where do they come from? For statevector evaluations such small number seem fine but for shot-based readout (or real hardware) they are probably too small.

Reply:
These values come from our current default value (and they come from scipy's default value). 
e.g. https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/optimizers/l_bfgs_b.py#L54, https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/optimizers/l_bfgs_b.py#L54, https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/optimizers/slsqp.py#L47

> For statevector evaluations such small number seem fine but for shot-based readout (or real hardware) they are probably too small.

This is right. I found our optimizers which is not target of this PR (e.g. ADAM) also do not work in default params on the situation with sampling errors due to same reason. At the moment, there is no standard for determining the default parameters, and I think it is generally nice to use the parameter shift rule.

Reply:
Ok, if this is what we currently have anyways I'm good  👍🏻